connect.go

```diff
@@ -61,32 +61,22 @@ func (b *Bundle) StartBackup(conn *pgx.Conn, backup string) (backupName string,
 	// We extract here version since it is not used elsewhere. If reused, this should be refactored.
 	// TODO: implement offline backups, incapsulate PostgreSQL version logic and create test specs for this logic.
 	// Currently all version-dependent logic is here
-	err = conn.QueryRow("select (current_setting('server_version_num'))::int").Scan(&version)
-	if err != nil {
-		return "", 0, version, errors.Wrap(err, "StartBackup: getting Postgres version failed")
-	}
-	walname := "xlog"
-	if version >= 100000 {
-		walname = "wal"
-	}
-
-	query := "SELECT case when pg_is_in_recovery() then '' else (pg_" + walname + "file_name_offset(lsn)).file_name end, lsn::text, pg_is_in_recovery() FROM pg_start_backup($1, true, false) lsn"
-	if err = conn.QueryRow(query, backup).Scan(&name, &lsnStr, &b.Replica); err != nil {
-		return "", 0, version, errors.Wrap(err, "StartBackup: pg_start_backup() failed")
-	}
+	queryBuilder := PgQueryBuilder{}
+	queryRunner := PgQueryRunner{queryBuilder: queryBuilder, connection: conn}
```

Comment:
maybe create a `NewPgQueryRunner` function so that you don't have to make an empty `queryBuilder`:

~~~go
func NewPgQueryRunner(conn *pgx.Conn) *PgQueryRunner {
    return &PgQueryRunner{queryBuilder:&PgQueryBuilder{},conn:conn} 
}
~~~

And actually maybe it should do GetVersion in the constructor and return an error:

~~~go
func NewPgQueryRunner(conn *pgx.Conn) (*PgQueryRunner, error) {
    r := &PgQueryRunner{queryBuilder:&PgQueryBuilder{},conn:conn} 
    var err error
    r.Version, err = r.getVersion() 
    if err != nil {
        return nil, err
    }
    return r, nil
}
~~~

And make `GetVersion` to be `getVersion` so its private.

Reply:
(and you'd reference `queryRunner.Version` instead of `queryBuilder.Version`?